### PR TITLE
feat(pipes-targets): add Kinesis

### PR DESCRIPTION
**Pull request by msambol**
Add Kinesis data stream as a Pipes target.

It's nontrivial to get the data from the Kinesis data stream, but here are screenshots 
showing data made it through during the integration test.

Screenshot 2024-06-24 at 7 24 10 PM

Screenshot 2024-06-24 at 7 26 35 PM

### CODE REVIEW

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Kinesis target properties.
+ */
+export interface KinesisTargetParameters {
+  /**
+   * The input transformation to apply to the message before sending it to the target.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipetargetparameters.html#cfn-pipes-pipe-pipetargetparameters-inputtemplate
+   * @default none
```
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts:13]:
```suggestion
   * @default - none
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   * Unicode strings with a maximum length limit of 256 characters for each key. Amazon Kinesis
+   * Data Streams uses the partition key as input to a hash function that maps the partition key
+   * and associated data to a specific shard. Specifically, an MD5 hash function is used to map
+   * partition keys to 128-bit integer values and to map associated data records to shards. As a
+   * result of this hashing mechanism, all data records with the same partition key map to the
+   * same shard within the stream.
+   *
+   * Minimum: 0
+   * Maximum: 256
+   *
```
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts:28]:
```suggestion
```

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  readonly inputTransformation?: IInputTransformation;
+
+  /**
+   * Determines which shard in the stream the data record is assigned to. Partition keys are
+   * Unicode strings with a maximum length limit of 256 characters for each key. Amazon Kinesis
+   * Data Streams uses the partition key as input to a hash function that maps the partition key
+   * and associated data to a specific shard. Specifically, an MD5 hash function is used to map
+   * partition keys to 128-bit integer values and to map associated data records to shards. As a
+   * result of this hashing mechanism, all data records with the same partition key map to the
+   * same shard within the stream.
```
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts:24]: This seems way too detailed, I don't think we should be explaining what Kinesis shards are here

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        kinesisStreamParameters: this.streamParameters,
+      },
+    };
+  }
+}
+
+function validatePartitionKey(pk: string) {
+  if (pk.length < 0 || pk.length > 256) {
+    throw new Error(`Partition key must be between 0 and 256 characters, received ${pk.length}`);
+  }
```
Comment (nmussy) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts:71]: Needless negative length check

```suggestion
  if (pk.length > 256) {
    throw new Error(`Partition key must be less or equal to 256 characters, received ${pk.length}`);
  }
```

**Review by nmussy — CHANGES_REQUESTED** [review]
Just a few nits, should be good to go otherwise

**Comment by msambol** [comment]
@nmussy Updated as well. Thanks again!

File: packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  /**
+   * Determines which shard in the stream the data record is assigned to.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipetargetkinesisstreamparameters.html#cfn-pipes-pipe-pipetargetkinesisstreamparameters-partitionkey
+   */
+  readonly partitionKey: string;
+}
+
+/**
+ * A EventBridge Pipes target that sends messages to a Kinesis stream.
```
Comment (Leo10Gama) [inline packages/@aws-cdk/aws-pipes-targets-alpha/lib/kinesis.ts:27]: nit:
```suggestion
 * An EventBridge Pipes target that sends messages to a Kinesis stream.
```

**Review by Leo10Gama — DISMISSED** [review]
Thanks for the PR! Just a few minor changes/suggestions from my end.

**Comment by msambol** [comment]
@Leo10Gama – fixed. Thanks for the review!

**Review by Leo10Gama — APPROVED** [review]
LGTM, thanks for the contribution!

**Comment by Leo10Gama** [comment]
@mergifyio update

**Pull request merged**